improver_tests/precipitation_type/snow_splitter/test_snow_splitter.py

```diff
@@ -0,0 +1,125 @@
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Tests for the SnowSplitter plugin"""
+import numpy as np
+import pytest
+from iris.cube import Cube, CubeList
+
+from improver.precipitation_type.snow_splitter import SnowSplitter
+from improver.synthetic_data.set_up_test_cubes import set_up_variable_cube
+
+LOCAL_MANDATORY_ATTRIBUTES = {
+    "title": "unit test data",
+    "source": "unit test",
+    "institution": "somewhere",
+}
+
+
+@pytest.fixture()
+def snow_cube() -> Cube:
+    """Set up a r, y, x cube of probability of snow at surface"""
+    data = np.full((2, 2, 2), fill_value=1, dtype=np.float32)
+    snow_phase_prob_cube = set_up_variable_cube(
+        data,
+        name="probability_of_snow_at_surface",
+        units="1",
+        attributes=LOCAL_MANDATORY_ATTRIBUTES,
+    )
+    return snow_phase_prob_cube
+
+
+@pytest.fixture()
+def rain_cube() -> Cube:
+    """Set up a r, y, x cube of probability of rain at surface"""
+    data = np.full((2, 2, 2), fill_value=0, dtype=np.float32)
+    rain_phase_prob_cube = set_up_variable_cube(
+        data,
+        name="probability_of_rain_at_surface",
+        units="1",
+        attributes=LOCAL_MANDATORY_ATTRIBUTES,
+    )
+    return rain_phase_prob_cube
+
+
+@pytest.fixture()
+def precip_rate_cube() -> Cube:
+    """Set up a r, y, x cube of precipitation rate"""
+    data = np.full((2, 2, 2), fill_value=1, dtype=np.float32)
+    precip_cube = set_up_variable_cube(
+        data,
+        name="precipitation_rate",
+        units="m/s",
+        attributes=LOCAL_MANDATORY_ATTRIBUTES,
+    )
+    return precip_cube
+
+
+@pytest.mark.parametrize(
+    "output_variable,cube_name", (("rain", "rain_rate"), ("snow", "snow_rate"))
+)
+@pytest.mark.parametrize(
+    "rain_value, snow_value,expected",
+    ((1, 1, 0.5), (0, 1, "dependent"), (1, 0, "dependent")),
+)
+def test_basic(
+    snow_cube,
+    rain_cube,
+    precip_rate_cube,
+    snow_value,
+    rain_value,
+    output_variable,
+    cube_name,
+    expected,
+):
+    """Check that for all possible combinations of rain and snow probabilities the correct
+    rain/snow rate is returned. The correct output will sometimes depend on what output_variable
+    is requested. Also check the name of the returned cube has been updated correctly"""
+    rain_cube.data = np.full_like(rain_cube.data, rain_value)
+    snow_cube.data = np.full_like(snow_cube.data, snow_value)
+
+    result = SnowSplitter(variable=output_variable)(
+        CubeList([snow_cube, rain_cube, precip_rate_cube])
+    )
+
+    if expected == "dependent":
+        expected = rain_value if output_variable == "rain" else snow_value
+
+    assert np.isclose(result.data, expected).all()
+    assert result.name() == cube_name
+
```

Comment:
Please can you check that the cube units are "m/s" and that the attributes match those on precip_rate_cube.

Reply:
Done